server/src/main/java/org/elasticsearch/search/aggregations/metrics/TopHitsAggregator.java

```diff
@@ -216,6 +217,26 @@ public InternalAggregation buildAggregation(long owningBucketOrdinal) throws IOE
         );
     }
 
+    private static FetchSearchResult runFetchPhase(SubSearchContext subSearchContext, int[] docIdsToLoad) {
+        // Fork the search execution context for each slice, because the fetch phase does not support concurrent execution yet.
+        SearchExecutionContext searchExecutionContext = new SearchExecutionContext(subSearchContext.getSearchExecutionContext()) {
+            @Override
+            public NestedDocuments getNestedDocuments() {
+                // this is a horrible hack: AggregationTestCase mocks SearchExecutionContext to override the resolution of nested documents.
+                // That gets lost as we fork the search execution context: overriding this method preserves that.
+                return subSearchContext.getSearchExecutionContext().getNestedDocuments();
```

Comment:
this is pretty pretty bad. I did not find a better way to make `TermsAggregatorTests` not fail. `AggregatorTestCase` has some weird setup that makes `SearchExecutionContext` return a `NestedDocuments` impl based on an empty `MappingLookup`, but that gets lost once the context is forked.

Reply:
@iverase kindly provided a patch that converts the two problematic tests to integration tests. This should be addressed now.

Reply:
Thanks @iverase! And thanks @javanna for fixing top_hits agg to work with parallel search!